### PR TITLE
Feature/1468 support for bulk quotes post passthrough

### PR DESCRIPTION
**Pull request by rmothilal**
added support for bulk quotes post functionality 
updated dependencies to fix audit issues

### CODE REVIEW

File: src/handlers/bulkQuotes/{id}.js [hunk 1]
```diff
@@ -32,30 +32,99 @@
...
+      const spanTags = LibUtil.getSpanTags(request, Enum.Events.Event.Type.BULK_QUOTE, Enum.Events.Event.Action.GET)
+      span.setTags(spanTags)
+      await span.audit({
+        headers: request.headers,
+        payload: request.payload
+      }, EventSdk.AuditEventAction.start)
+      // call the model to re-forward the quote update to the correct party
+      // note that we do not check if our caller is the correct party, but we
+      // will send the callback to the correct party regardless.
+      const result = await model.handleBulkQuoteGet(request.headers, bulkQuoteId, span)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}.js:77]: `await` should not be necessary except of course for the logging.

File: src/handlers/bulkQuotes/{id}.js [hunk 2]
```diff
@@ -32,30 +32,99 @@
...
+      request.server.log(['info'], `GET bulkQuotes/{id} request succeeded and returned: ${util.inspect(result)}`)
+    } catch (err) {
+      // something went wrong, use the model to handle the error in a sensible way
+      request.server.log(['error'], `ERROR - GET /bulkQuotes/{id}: ${LibUtil.getStackOrInspect(err)}`)
+      await model.handleException(fspiopSource, bulkQuoteId, err, request.headers, span)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}.js:82]: `await` should not be necessary.

File: src/handlers/bulkQuotes/{id}.js [hunk 3]
```diff
@@ -32,30 +32,99 @@
...
+        headers: request.headers,
+        payload: request.payload
+      }, EventSdk.AuditEventAction.start)
+      // call the quote update handler in the model
+      const result = await model.handleBulkQuoteUpdate(request.headers, bulkQuoteId, request.payload, span)
+      request.server.log(['info'], `PUT bulk quote request succeeded and returned: ${util.inspect(result)}`)
+    } catch (err) {
+      // something went wrong, use the model to handle the error in a sensible way
+      request.server.log(['error'], `ERROR - PUT /bulkQuotes/{id}: ${LibUtil.getStackOrInspect(err)}`)
+      await model.handleException(fspiopSource, bulkQuoteId, err, request.headers, span)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}.js:124]: `await` should not be necessary.

File: src/handlers/bulkQuotes/{id}.js [hunk 4]
```diff
@@ -32,30 +32,99 @@
...
+    const span = request.span
+    try {
+      const spanTags = LibUtil.getSpanTags(request, Enum.Events.Event.Type.BULK_QUOTE, Enum.Events.Event.Action.FULFIL)
+      span.setTags(spanTags)
+      await span.audit({
+        headers: request.headers,
+        payload: request.payload
+      }, EventSdk.AuditEventAction.start)
+      // call the quote update handler in the model
+      const result = await model.handleBulkQuoteUpdate(request.headers, bulkQuoteId, request.payload, span)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}.js:119]: `await` should not be necessary except for the logging.

File: src/handlers/bulkQuotes/{id}/error.js [hunk 5]
```diff
@@ -45,7 +48,39 @@ module.exports = {
...
+    const span = request.span
+    try {
+      const spanTags = LibUtil.getSpanTags(request, Enum.Events.Event.Type.BULK_QUOTE, Enum.Events.Event.Action.ABORT)
+      span.setTags(spanTags)
+      await span.audit({
+        headers: request.headers,
+        payload: request.payload
+      }, EventSdk.AuditEventAction.start)
+      // call the quote error handler in the model
+      const result = await model.handleBulkQuoteError(request.headers, bulkQuoteId, request.payload.errorInformation, span)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}/error.js:75]: `await` should not be necessary except for the logging.

File: src/handlers/bulkQuotes/{id}/error.js [hunk 6]
```diff
@@ -45,7 +48,39 @@ module.exports = {
...
+      request.server.log(['info'], `PUT bulkQuote error request succeeded and returned: ${util.inspect(result)}`)
+    } catch (err) {
+      // something went wrong, use the model to handle the error in a sensible way
+      request.server.log(['error'], `ERROR - PUT /bulkQuotes/{id}/error: ${LibUtil.getStackOrInspect(err)}`)
+      await model.handleException(fspiopSource, bulkQuoteId, err, request.headers)
```
Comment (oderayi) [inline src/handlers/bulkQuotes/{id}/error.js:80]: `await` should not be necessary.

File: src/model/bulkQuotes.js [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+        throw ErrorHandler.CreateFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.VALIDATION_ERROR,
+          `Update for bulk quote ${bulkQuoteId} failed: "accept" header should not be sent in callbacks.`, null, headers['fspiop-source'])
+      }
+      // accumulate enum ids
+      const refs = {}
+      // if we got here rules passed, so we can forward the quote on to the recipient dfsp
+      const childSpan = span.getChild('qs_quote_forwardBulkQuoteUpdate')
+      try {
+        await childSpan.audit({ headers, params: { bulkQuoteId }, payload: bulkQuoteUpdateRequest }, EventSdk.AuditEventAction.start)
+        await this.forwardBulkQuoteUpdate(headers, bulkQuoteId, bulkQuoteUpdateRequest, childSpan)
```
Comment (oderayi) [inline src/model/bulkQuotes.js:204]: `await` might not be necessary.

File: src/model/bulkQuotes.js [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+      } catch (err) {
+        // any-error
+        // as we are on our own in this context, dont just rethrow the error, instead...
+        // get the model to handle it
+        const fspiopSource = headers[ENUM.Http.Headers.FSPIOP.SOURCE]
+        this.writeLog(`Error forwarding bulk quote update: ${getStackOrInspect(err)}. Attempting to send error callback to ${fspiopSource}`)
+        await this.handleException(fspiopSource, bulkQuoteId, err, headers, childSpan)
```
Comment (oderayi) [inline src/model/bulkQuotes.js:211]: `await` might not be necessary.

File: src/model/bulkQuotes.js [hunk 9]
```diff
@@ -0,0 +1,556 @@
...
+      }
+
+      // all ok, return refs
+      return refs
+    } catch (err) {
+      // internal-error
+      this.writeLog(`Error in handleBulkQuoteUpdate: ${getStackOrInspect(err)}`)
+      if (txn) {
+        txn.rollback(err)
+      }
```
Comment (oderayi) [inline src/model/bulkQuotes.js:225]: `txn` is not assigned.

File: src/model/bulkQuotes.js [hunk 10]
```diff
@@ -0,0 +1,556 @@
...
+    }
+  }
+
+  /**
+   * Logic for handling quote update requests e.g. PUT /bulkQuotes/{id} requests
+   *
+   * @returns {object} - object containing updated entities
+   */
+  async handleBulkQuoteUpdate (headers, bulkQuoteId, bulkQuoteUpdateRequest, span) {
+    const txn = null
```
Comment (oderayi) [inline src/model/bulkQuotes.js:189]: can be removed.

File: src/model/bulkQuotes.js [hunk 11]
```diff
@@ -0,0 +1,556 @@
...
+   *
+   * @returns {undefined}
+   */
+  async handleBulkQuoteGet (headers, bulkQuoteId, span) {
+    const fspiopSource = headers[ENUM.Http.Headers.FSPIOP.SOURCE]
+    try {
+      const childSpan = span.getChild('qs_quote_forwardBulkQuoteGet')
+      try {
+        await childSpan.audit({ headers, params: { bulkQuoteId } }, EventSdk.AuditEventAction.start)
+        await this.forwardBulkQuoteGet(headers, bulkQuoteId, childSpan)
```
Comment (oderayi) [inline src/model/bulkQuotes.js:299]: `await` might not be necessary.

File: src/model/bulkQuotes.js [hunk 12]
```diff
@@ -0,0 +1,556 @@
...
+      } catch (err) {
+        // any-error
+        // as we are on our own in this context, dont just rethrow the error, instead...
+        // get the model to handle it
+        this.writeLog(`Error forwarding bulk quote get: ${getStackOrInspect(err)}. Attempting to send error callback to ${fspiopSource}`)
+        await this.handleException(fspiopSource, bulkQuoteId, err, headers, childSpan)
```
Comment (oderayi) [inline src/model/bulkQuotes.js:305]: `await` might not be necessary.

File: src/model/bulkQuotes.js (hunk 8 above)
Comment (rmothilal) [reply src/model/bulkQuotes.js:211]: in the model layer it is necessary

File: src/model/bulkQuotes.js (hunk 10 above)
Comment (rmothilal) [reply src/model/bulkQuotes.js:189]: in the model layer it is necessary

File: src/model/bulkQuotes.js (hunk 11 above)
Comment (rmothilal) [reply src/model/bulkQuotes.js:299]: in the model layer it is necessary

File: src/model/bulkQuotes.js (hunk 12 above)
Comment (rmothilal) [reply src/model/bulkQuotes.js:305]: in the model layer it is necessary

File: src/model/bulkQuotes.js (hunk 7 above)
Comment (rmothilal) [reply src/model/bulkQuotes.js:204]: in the model layer it is necessary

**Pull request merged**